lib/capybara/node/finders.rb

```diff
@@ -166,7 +166,13 @@ def all(*args)
       # @return [Capybara::Node::Element]            The found element or nil
       #
       def first(*args)
+        if Capybara.first_default_waiting
+          options = if args.last.is_a?(Hash) then args.pop.dup else {} end
+          args.push({minimum: 1}.merge(options))
+        end
         all(*args).first
+      rescue Capybara::ExpectationNotMet
+        nil
```

Comment:
Ok, you make a good point, and it will be considered for v3 but for now we need to stick with matching current behavior of first() and the documented description of #first, as @abotalov pointed out you can do.  find(:link, link, match: :first).click  if you want a method that will raise